src/XMakeBuildEngine/Definition/ToolsetReader.cs

```diff
@@ -157,89 +157,86 @@ ToolsetDefinitionLocations locations
-                        foreach (var d in Directory.GetDirectories(xbuildToolsetsDir).Where(d => r.IsMatch(d)))
-                        {
-                            var version = Path.GetFileName(d);
-                            var binPath = Path.Combine(d, "bin");
-                            if (version != null && !toolsets.ContainsKey(version))
-                            {
-                                // Create standard properties. On Mono they are well known
-                                PropertyDictionary<ProjectPropertyInstance> buildProperties =
-                                    CreateStandardProperties(globalProperties, version, xbuildToolsetsDir, binPath);
-
-                                toolsets.Add(
-                                    version,
-                                    new Toolset(
-                                        version,
-                                        binPath,
-                                        buildProperties,
-                                        environmentProperties,
-                                        globalProperties,
-                                        null,
-                                        currentDir,
-                                        string.Empty));
-                            }
-                        }
+                                v4Dir,
+                                buildProperties,
+                                environmentProperties,
+                                globalProperties,
+                                null,
+                                currentDir,
+                                string.Empty));
+                    }
 
-                        if (!toolsets.ContainsKey(MSBuildConstants.CurrentToolsVersion))
-                        {
-                            toolsets.Add(
-                                MSBuildConstants.CurrentToolsVersion,
-                                new Toolset(
-                                    MSBuildConstants.CurrentToolsVersion,
-                                    currentDir,
-                                    props,
-                                    new PropertyDictionary<ProjectPropertyInstance>(),
-                                    currentDir,
-                                    string.Empty));
-                        }
+                    // Other toolsets are installed in the xbuild directory
+                    var xbuildToolsetsDir = Path.Combine(libraryPath, "xbuild");
+                    var r = new Regex(xbuildToolsetsDir + Path.DirectorySeparatorChar + @"\d+\.\d+");
+                    foreach (var d in Directory.GetDirectories(xbuildToolsetsDir).Where(d => r.IsMatch(d)))
+                    {
+                        var version = Path.GetFileName(d);
+                        var binPath = Path.Combine(d, "bin");
+                        if (toolsets.ContainsKey(version))
```

Comment:
This change here was easy to miss, because of the big diff and most of it being just "indentation change". This commit is fixing the "merge" issue, so, it should not contain these unrelated changes. It can be fixed in the separate commit where you are fixing the Regex issue.